lightsim2grid/newtonpf/newtonpf.py

```diff
@@ -306,14 +307,16 @@ def newtonpf_new(Ybus, Sbus, V0, ref, pv, pq, ppci, options):
 
     # initialize the solver
     # TODO have that in options maybe (can use GaussSeidel, and NR with KLU -faster- or SparseLU)
-    if KLU_solver_available:
-        solver = KLUSolver()
+    if options["distributed_slack"]:
+        solver = KLUSolver() if KLU_solver_available else SparseLUSolver()
     else:
-        solver = SparseLUSolver()
-    Ybus = sparse.csc_matrix(Ybus)
+        solver = KLUSolverSingleSlack() if KLU_solver_available else SparseLUSolverSingleSlack()
```

Comment:
Looks good ! It's more consistent :-)

Will it work with earlier version of pandapower ? (maybe a check that `"distributed_slack" in options` should be added I think

Reply:
I will modify the code to account for when distributed_slack is not in options